### PR TITLE
Fixes required fields in server group modal. 

### PR DESCRIPTION
**Pull request by danielpeach**
Fixes required fields in server group modal.
The UI did not indicate that the wizard was incomplete if zone was left unselected.
 
@lwander @duftler PTAL

### CODE REVIEW

File: app/scripts/modules/google/serverGroup/configure/serverGroupConfiguration.service.js [hunk 1]
```diff
@@ -387,7 +387,7 @@ module.exports = angular.module('spinnaker.serverGroup.configure.gce.configurati
         } else {
           filteredData.zones = null;
         }
-
+
```
Comment (lwander) [inline app/scripts/modules/google/serverGroup/configure/serverGroupConfiguration.service.js]: Extraneous whitespace

File: app/scripts/modules/google/serverGroup/configure/wizard/cloneServerGroup.gce.controller.js [hunk 2]
```diff
@@ -121,6 +123,11 @@ module.exports = angular.module('spinnaker.serverGroup.configure.gce.cloneServer
       if (result.dirty.availabilityZones) {
         v2modalWizardService.markDirty('capacity');
       }
+      if(result.dirty.zone){
```
Comment (lwander) [inline app/scripts/modules/google/serverGroup/configure/wizard/cloneServerGroup.gce.controller.js]: Spaces missing before `(` and after `)`

File: app/scripts/modules/google/serverGroup/configure/serverGroupConfiguration.service.js [hunk 3]
```diff
@@ -449,6 +449,17 @@ module.exports = angular.module('spinnaker.serverGroup.configure.gce.configurati
 
         return result;
       };
+
+      command.zoneChanged = function zoneChanged() {
+        var result = { dirty: { } };
+        if(command.zone === undefined) {
```
Comment (lwander) [inline app/scripts/modules/google/serverGroup/configure/serverGroupConfiguration.service.js]: Missing space before `(`

**Comment by lwander** [comment]
LGTM, just a few style comments

File: app/scripts/modules/google/serverGroup/configure/serverGroupConfiguration.service.js (hunk 3 above)
Comment (duftler) [inline app/scripts/modules/google/serverGroup/configure/serverGroupConfiguration.service.js]: I believe this also has to consider `command.regional`. If the server group being configured is regional, no zone is required.

**Comment by anotherchrisberry** [comment]
LGTM - needs a rebase

**Pull request merged**